### PR TITLE
Resolve 3.4.1

### PR DESCRIPTION
**Pull request by slubwama**
## Issue I worked on

**Issue 3.4.1**

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can we combine this pull request with modifying this file https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/webapp/uncaughtException.jsp to simply display an Authentication is required error message for cases when one has not yet logged in?

**Comment by slubwama** [comment]
> Can we combine this pull request with modifying this file https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/webapp/uncaughtException.jsp to simply display an Authentication is required error message for cases when one has not yet logged in? That way, we do not expose the fact that we did more than simply check for authentication.

Makes Sense will make a second Pr in Legacy UI

File: web/src/main/resources/openmrs-servlet.xml [hunk 1]
```diff
@@ -91,7 +91,8 @@
  	<bean class="org.springframework.web.servlet.handler.SimpleMappingExceptionResolver"> 
  		<property name="exceptionMappings"> 
  			<map> 
- 				<entry key="org.openmrs.api.APIException" value="uncaughtException"/> 
+ 				<entry key="org.openmrs.api.APIException" value="uncaughtException"/>
+				<entry key="org.springframework.web.method.annotation.MethodArgumentTypeMismatchException" value="uncaughtException"/>
```
Comment (dkayiwa) [inline web/src/main/resources/openmrs-servlet.xml:95]: What do we do for other exceptions which are not `MethodArgumentTypeMismatchException`?

File: web/src/main/resources/openmrs-servlet.xml [hunk 2]
```diff
@@ -91,7 +91,9 @@
  	<bean class="org.springframework.web.servlet.handler.SimpleMappingExceptionResolver"> 
  		<property name="exceptionMappings"> 
  			<map> 
- 				<entry key="org.openmrs.api.APIException" value="uncaughtException"/> 
+ 				<entry key="org.openmrs.api.APIException" value="uncaughtException"/>
+				<entry key="org.springframework.core.NestedRuntimeException" value="uncaughtException"/>
```
Comment (dkayiwa) [inline web/src/main/resources/openmrs-servlet.xml:95]: What is the value of having both `org.springframework.core.NestedRuntimeException` and `java.lang.Exception`?

Comment (slubwama) [reply web/src/main/resources/openmrs-servlet.xml]: @dkayiwa resolved

**Pull request merged**